Add test method TestPinMessageInCommunityChat

shinnok opened the pull request:
Add TestPinMessageInCommunityChat to test that a pin message type can be sent correctly for community\chat members or denied for non-members.

~~Need to fix `community.AdminSettings().PinMessageAllMembersEnabled` setting first, which seems to be ignored for non-owner members.~~

Closes #4138

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (42)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 031fff90 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/1/) | 2023-10-11 10:17:48 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231011-101403-031fff-pr4139.zip) |
| :heavy_check_mark: | 031fff90 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/1/) | 2023-10-11 10:18:59 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231011-101403-031fff-pr4139.zip) |
| :heavy_check_mark: | 031fff90 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/1/) | 2023-10-11 10:19:32 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231011-101403-031fff-pr4139.aar) |
| :heavy_multiplication_x: | 031fff90 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/1/) | 2023-10-11 10:30:45 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 88ec6b49 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/2/) | 2023-10-13 13:55:04 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-135241-88ec6b-pr4139.aar) |
| :heavy_check_mark: | 88ec6b49 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/2/) | 2023-10-13 13:55:40 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-135241-88ec6b-pr4139.zip) |
| :heavy_check_mark: | 88ec6b49 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/2/) | 2023-10-13 13:56:02 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-135241-88ec6b-pr4139.zip) |
| :heavy_multiplication_x: | 88ec6b49 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/2/) | 2023-10-13 14:08:39 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 95f97ec5 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/3/) | 2023-10-13 14:21:29 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-142000-95f97e-pr4139.zip) |
| :heavy_check_mark: | 95f97ec5 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/3/) | 2023-10-13 14:21:32 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-141956-95f97e-pr4139.aar) |
| :heavy_check_mark: | 95f97ec5 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/3/) | 2023-10-13 14:22:50 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-142000-95f97e-pr4139.zip) |
| :heavy_multiplication_x: | 95f97ec5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/3/) | 2023-10-13 14:35:45 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/3//consoleText) |
| :heavy_check_mark: | 95f97ec5 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/4/) | 2023-10-17 06:38:12 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-063703-95f97e-pr4139.zip) |
| :heavy_check_mark: | 95f97ec5 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/4/) | 2023-10-17 06:38:39 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-063703-95f97e-pr4139.aar) |
| :heavy_check_mark: | 95f97ec5 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/4/) | 2023-10-17 06:40:04 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-063703-95f97e-pr4139.zip) |
| :heavy_multiplication_x: | 95f97ec5 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/4/) | 2023-10-17 07:05:24 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/4//consoleText) |
| :heavy_multiplication_x: | 95f97ec5 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/5/) | 2023-10-17 07:19:26 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ae6aa0bf | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/5/) | 2023-10-17 08:40:29 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-083824-ae6aa0-pr4139.aar) |
| :heavy_check_mark: | ae6aa0bf | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/5/) | 2023-10-17 08:41:40 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-083824-ae6aa0-pr4139.zip) |
| :heavy_check_mark: | ae6aa0bf | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/5/) | 2023-10-17 08:42:10 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-083824-ae6aa0-pr4139.zip) |
| :heavy_multiplication_x: | ae6aa0bf | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/6/) | 2023-10-17 08:51:31 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0745ecf8 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/6/) | 2023-10-23 17:40:53 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231023-173744-0745ec-pr4139.zip) |
| :heavy_check_mark: | 0745ecf8 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/6/) | 2023-10-23 17:41:46 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231023-173744-0745ec-pr4139.zip) |
| :heavy_check_mark: | 0745ecf8 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/6/) | 2023-10-23 17:42:56 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231023-173744-0745ec-pr4139.aar) |
| :heavy_multiplication_x: | 0745ecf8 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/7/) | 2023-10-23 17:53:16 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4cc74dde | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/7/) | 2023-10-23 17:47:29 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231023-174414-4cc74d-pr4139.zip) |
| :heavy_check_mark: | 4cc74dde | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/7/) | 2023-10-23 17:48:01 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231023-174414-4cc74d-pr4139.zip) |
| :heavy_check_mark: | 4cc74dde | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/7/) | 2023-10-23 17:48:46 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231023-174414-4cc74d-pr4139.aar) |
| :heavy_multiplication_x: | 4cc74dde | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/8/) | 2023-10-23 17:55:27 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 56c357da | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/8/) | 2023-10-23 18:24:31 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231023-182134-56c357-pr4139.zip) |
| :heavy_check_mark: | 56c357da | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/8/) | 2023-10-23 18:25:03 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231023-182134-56c357-pr4139.zip) |
| :heavy_multiplication_x: | 56c357da | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/9/) | 2023-10-23 18:25:21 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/9//consoleText) |
| :heavy_check_mark: | 56c357da | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/9/) | 2023-10-23 18:25:52 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231023-182437-56c357-pr4139.zip) |
| :heavy_check_mark: | 56c357da | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/8/) | 2023-10-23 18:26:57 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231023-182134-56c357-pr4139.aar) |
| :heavy_check_mark: | 56c357da | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/9/) | 2023-10-23 18:28:18 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231023-182506-56c357-pr4139.zip) |
| :heavy_check_mark: | 56c357da | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/9/) | 2023-10-23 18:32:32 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231023-182700-56c357-pr4139.aar) |
| :heavy_check_mark: | 56c357da | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/10/) | 2023-10-23 18:56:47 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/10//consoleText) |
| :heavy_check_mark: | 56c357da | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/11/) | 2023-10-24 17:29:35 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ebbb54fb | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/10/) | 2023-10-25 13:50:25 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231025-134836-ebbb54-pr4139.zip) |
| :heavy_check_mark: | ebbb54fb | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/10/) | 2023-10-25 13:50:43 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231025-134832-ebbb54-pr4139.aar) |
| :heavy_multiplication_x: | ebbb54fb | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/12/) | 2023-10-25 13:50:57 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/12//consoleText) |
| :heavy_check_mark: | ebbb54fb | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/10/) | 2023-10-25 13:52:35 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231025-134836-ebbb54-pr4139.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 2bc6774e | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/11/) | 2023-10-25 14:02:37 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231025-140112-2bc677-pr4139.zip) |
| :heavy_check_mark: | 2bc6774e | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/11/) | 2023-10-25 14:03:22 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231025-140107-2bc677-pr4139.aar) |
| :heavy_multiplication_x: | 2bc6774e | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/13/) | 2023-10-25 14:03:50 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/13//consoleText) |
| :heavy_check_mark: | 2bc6774e | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/11/) | 2023-10-25 14:05:02 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231025-140112-2bc677-pr4139.zip) |
| :heavy_multiplication_x: | 2bc6774e | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/14/) | 2023-10-25 14:19:40 | ~12 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/14//consoleText) |
| :heavy_multiplication_x: | 2bc6774e | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/15/) | 2023-10-25 14:36:50 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/15//consoleText) |
| :heavy_multiplication_x: | 2bc6774e | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/16/) | 2023-10-25 17:48:42 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/16//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 124bc3e5 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4139/12/) | 2023-10-25 18:12:58 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231025-181108-124bc3-pr4139.zip) |
| :heavy_check_mark: | 124bc3e5 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4139/12/) | 2023-10-25 18:13:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231025-181104-124bc3-pr4139.aar) |
| :heavy_multiplication_x: | 124bc3e5 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/17/) | 2023-10-25 18:14:01 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/17//consoleText) |
| :heavy_check_mark: | 124bc3e5 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4139/12/) | 2023-10-25 18:14:43 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231025-181107-124bc3-pr4139.zip) |
| :heavy_multiplication_x: | 124bc3e5 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/18/) | 2023-10-25 18:49:53 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/18//consoleText) |
| :heavy_multiplication_x: | 124bc3e5 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/19/) | 2023-10-25 18:56:06 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/19//consoleText) |
| :heavy_multiplication_x: | 124bc3e5 | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/20/) | 2023-10-26 06:42:08 | ~17 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/20//consoleText) |
| :heavy_check_mark: | 124bc3e5 | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/21/) | 2023-10-26 07:29:12 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/21//consoleText) |
| :heavy_multiplication_x: | 124bc3e5 | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/22/) | 2023-10-26 07:43:05 | ~6 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/22//consoleText) |
| :heavy_multiplication_x: | 124bc3e5 | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/23/) | 2023-10-26 08:49:20 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/23//consoleText) |
| :heavy_check_mark: | 124bc3e5 | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/24/) | 2023-10-26 09:36:07 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4139/24//consoleText) |

shinnok commented on protocol/communities_messenger_test.go:
```diff
@@ -495,6 +495,42 @@ func (s *MessengerCommunitiesSuite) TestPostToCommunityChat() {
+	s.Require().NoError(err)
+	s.Require().Len(sendResponse.Messages(), 1)
+
+	community, err = s.admin.communitiesManager.GetByID(community.ID())
```
Grabbing the community again, either via s.admin or via s.[alice|bob] doesn't not yield the CommunityAdminSettings member which was set initially at community creation.

shinnok committed:
Add TestPinMessageInCommunityChat, adjust pin permissions

Also check pin post permission a bit earlier for the sender, with
canPost() in sendPinMessage(), even though there is a later check down
the line in dispatchMessage().

Fixes #4138

shinnok commented:
To be merged after https://github.com/status-im/status-go/pull/4064 .

jrainville commented on protocol/communities_messenger_test.go:
```diff
@@ -484,6 +484,74 @@ func (s *MessengerCommunitiesSuite) TestPostToCommunityChat() {
+
+	// alice did not join the community, so she should not be able to send the pin message
+	s.advertiseCommunityTo(community, s.alice)
+	sendResponse, err = s.alice.SendPinMessage(ctx, pinMessage)
```
This needs Alice to Spectate the community too, so she has access to the Chat.

Also, I recommend creating a new `pinMessage`, as sending the same message with the same ID could cause a clash.

Finally, you can validate that the error message is the one you expect at the end, something like "user can't post".

shinnok replied:
Done. With the latest change, this should fail in the CI until #4064 counterpart is merged to develop.
